src/channel.ts

```diff
@@ -955,17 +956,21 @@ export class Channel<
       state: false,
       presence: false,
     };
-    return await this.query(options);
+    return await this.query(options, 'latest');
   };
 
   /**
    * query - Query the API, get messages, members or other channel fields
    *
    * @param {ChannelQueryOptions<ChannelType, CommandType, UserType>} options The query options
+   * @param {MessageSetType} messageSetToAddToIfDoesNotExist Which message set to add to if messages are not in state
    *
    * @return {Promise<ChannelAPIResponse<AttachmentType, ChannelType, CommandType, MessageType, ReactionType, UserType>>} Returns a query response
    */
-  async query(options: ChannelQueryOptions<ChannelType, CommandType, UserType>) {
+  async query(
+    options: ChannelQueryOptions<ChannelType, CommandType, UserType>,
+    messageSetToAddToIfDoesNotExist: MessageSetType = 'current',
```

Comment:
I am a little unclear about the meaning of this param. Would really appreciate if you can add some details about it in documentation. Example would be even better.

Reply:
Done, let me know if think it needs more clarification